rosbag2_cpp/include/rosbag2_cpp/readers/sequential_reader.hpp

```diff
@@ -127,13 +126,15 @@ class ROSBAG2_CPP_PUBLIC SequentialReader
     const std::string & storage_serialization_format);
 
   std::unique_ptr<rosbag2_storage::StorageFactoryInterface> storage_factory_{};
-  std::shared_ptr<SerializationFormatConverterFactoryInterface> converter_factory_{};
   std::shared_ptr<rosbag2_storage::storage_interfaces::ReadOnlyInterface> storage_{};
   std::unique_ptr<Converter> converter_{};
   std::unique_ptr<rosbag2_storage::MetadataIo> metadata_io_{};
   rosbag2_storage::BagMetadata metadata_{};
   std::vector<std::string> file_paths_{};  // List of database files.
   std::vector<std::string>::iterator current_file_iterator_{};  // Index of file to read from
+
+private:
+  std::shared_ptr<SerializationFormatConverterFactoryInterface> converter_factory_{};
```

Comment:
from all the things which moved into the protected section, why not the `converter_factory`? I am not proposing otherwise, just asking for the rationale.

Reply:
Every other previously-private member is currently referenced directly in the SequentialCompressionReader subclass. We can probably clean it up a little so that more can be private, but my intention with this was to go for the minimum necessary set of visibility increase. Which happened to just be everything except this